src/main/java/org/ays/auth/port/impl/AysRoleAdapter.java

```diff
@@ -38,6 +40,18 @@ class AysRoleAdapter implements AysRoleReadPort, AysRoleSavePort {
     private final AysRoleEntityToDomainMapper roleEntityToDomainMapper = AysRoleEntityToDomainMapper.initialize();
 
 
+    /**
+     * Retrieves all {@link AysRole} from the repository.
+     *
+     * @return A set of all {@link AysRole}.
+     */
+    @Override
+    public Set<AysRole> findAll() {
+        List<AysRoleEntity> roleEntities = roleRepository.findAll();
+        return new HashSet<>(roleEntityToDomainMapper.map(roleEntities));
+    }
```

Comment:
Set dönmek yerine `List` dönebiliriz. MapStruct kütüphanesinde fark ettiğimiz bir problem vardı. Bu sebeple spesifik ihtiyacımız olmadığı sürece `Set` yerine `List` kullanmamız daha sağlıklı olabilir.

```suggestion
     * @return A list of all {@link AysRole}.
     */
    @Override
    public List<AysRole> findAll() {
        List<AysRoleEntity> roleEntities = roleRepository.findAll();
        return roleEntityToDomainMapper.map(roleEntities);
    }
```

Reply:
değişikliği yaptım